addons/crm/models/res_partner.py

```diff
@@ -39,7 +40,8 @@ def default_get(self, fields):
     def _compute_opportunity_count(self):
         for partner in self:
             operator = 'child_of' if partner.is_company else '='  # the opportunity count should counts the opportunities of this company and all its contacts
-            partner.opportunity_count = self.env['crm.lead'].search_count([('partner_id', operator, partner.id), ('type', '=', 'opportunity')])
+            partner.partner_opportunity_ids = self.env['crm.lead'].search([('partner_id', operator, partner.id), ('type', '=', 'opportunity')])
```

Comment:
```suggestion
            # We use opportunity_count_ids instead of opportunity_ids since the 'name_search' method of the res.partner
            # is overridden to include more results.
            partner.opportunity_count_ids = self.env['crm.lead'].search([('partner_id', operator, partner.id), ('type', '=', 'opportunity')])
```